### PR TITLE
CRM-19379: ensure ACL restricted users have access to their own groups

### PR DESCRIPTION
**Pull request by jmcclelland**
- [CRM-19379: ACLs restricted users should have access to their own groups](https://issues.civicrm.org/jira/browse/CRM-19379)

### CODE REVIEW

File: CRM/ACL/BAO/ACL.php [hunk 1]
```diff
@@ -954,6 +954,12 @@ public static function group(
       $ids = $includedGroups;
     }
 
+    // Add groups this user owns.
+    $sql = "SELECT id FROM civicrm_group WHERE created_id = %0";
+    $dao = CRM_Core_DAO::executeQuery($sql, array(0 => array($contactID, 'Integer')));
+    while ($dao->fetch()) {
+      if (!in_array($dao->id, $ids)) $ids[] = $dao->id;
```
Comment (seamuslee001) [inline CRM/ACL/BAO/ACL.php]: @jmcclelland you need to add { around $ids[] = $dao->id; }

Comment (jmcclelland) [inline CRM/ACL/BAO/ACL.php]: Right! thanks - done.

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by eileenmcnaughton** [comment]
This won't merge any more due to another commit I did which gives permission to view hidden groups created by the contact.

I had some additional reservations which I noted on the JIRA with a couple of thoughts.

I'm going to close this now as it needs to be reworked to cope with subsequent merges. Please open a new PR when you are ready

**Pull request closed without merging**